app/models/portfolio.rb

```diff
@@ -16,13 +19,32 @@ class Portfolio < ApplicationRecord
   validates :enabled_before_type_cast, :format => { :with => /\A(true|false)\z/i }, :allow_blank => true
 
   has_many :portfolio_items, :dependent => :destroy
+  has_one  :ancillary_metadata, :as => :resource, :dependent => :destroy
 
   def add_portfolio_item(portfolio_item)
     portfolio_items << portfolio_item
   end
 
   def metadata
-    {:user_capabilities => user_capabilities,
-     :shared            => self.access_control_entries.any?}
+    # Merge dynamic user_capabilities properties
+    ancillary_metadata.attributes.except!('id', 'tenant_id', 'resource_id', 'resource_type', 'created_at')
```

Comment:
Think we can just do something like this?:
```
# in class AncillaryMetadata
def attributes
  super(:except => %w[id tenant_id resource_id resource_type created_at])
end
```

Then if you wanted we could make that array a constant as well.

Reply:
:+1: I like this idea, it keeps the custom `attributes` logic in the `AncillaryMetadata` class.

Reply:
@eclarizio I like this as well and did not the keys into a constant in the new model, but thought there might be issues with overriding the Rails `attributes` method and decide to use a new `metadata_attributes` method instead.